### PR TITLE
[Feature](mlu-ops): delete 200 code and adapt to toolkit 4.0

### PR DESCRIPTION
**Pull request by DanieeelLiu**
Thanks for your contribution and we appreciate it a lot. :rocket::rocket:

## 1. Motivation

Please describe your motivation and the goal you want to achieve through this pull request.

## 2. Modification

Please briefly describe what modification is made in this pull request, and indicate where to make the modification.

Are new test cases added? If so, please post the corresponding generator-PR link here.

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS™ Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- static threshold
  - diff1
    - [ ] float32 mlu diff1 <= 1e-5
    - [ ] float32 mlu diff1 <= 3e-3
    - [ ] float16 mlu diff1 <= 3e-3
  - diff2
    - [ ] float32 mlu diff2 <= 1e-5
    - [ ] float32 mlu diff2 <= 3e-3
    - [ ] float16 mlu diff2 <= 3e-3
  - diff3
    - [ ] mlu diff3 == 0
    - [ ] mlu diff3_1 == 0
    - [ ] mlu diff3_2 == 0
- dynamic threshold
  - [ ] diff1: mlu diff1 <= max(baseline diff1 * 10, static threshold)
  - [ ] diff2: mlu diff2 <= max(baseline diff2 * 10, static threshold)
  - [ ] diff3: mlu diff3 <= max(baseline diff3 * 10, static threshold)
    - float32, threshold = 1e-5
    - float16, threshold = 1e-3

#### 3.1.2 Operator Scheme checklist

- Supported hardware
  - [ ] MLU370
  - [ ] MLU590
- Job types
  - [ ] BLOCK
  - [ ] UNION1
  - [ ] UNION2
  - [ ] UNION4
  - [ ] The operator will dynamically select the most suitable task type, for example, UNION8

#### 3.2.1 Accuracy Test

If you have checked the following items, please tick the relevant box.

- [ ] Data type test (e.g. float32/int8)
- [ ] Multi-dimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [ ] stability test
- [ ] Multiple platform test
- [ ] Gen_case module test, see: [Gencase-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/Gencase-User-Guide-zh.md)
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [ ] For memory leak check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md)
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md)
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS™-Performance-Acceptance-Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md)

#### 3.2.2 Parameter Check

Test Point-1: `When a new operator is submitted, the test points are given and the test results are stated`. Acceptance Standard: `Normal error`.
```bash
Please fill your test results(Error Message) in here, ...
```

Test Point-2: `Whether illegal parameters are passed`. Acceptance Standard: `Normal error`.
```bash
Test results...
```

### 3.3 Performance Test

See [MLU-OPS™ Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

```bash
# The test results should contain Op name, Shape, Data type,  
#   MLU Hardware Time(us), MLU Interface Time(us), MLU IO Efficiency, 
#   MLU Compute Efficiency, and Mlu Workspace Size(Bytes)
# 
# for example:
#
# ----------- case0 -----------
# case0
# [Op name                ]: abs
# [Shape                  ]: input.shape=[1024,1024,3,4], output.shape=[1024,1024,3,4]
# [Data type]             ]: float32
# [MLU Hardware Time      ]: 15728 (us)
# [MLU Interface Time     ]: 369.008 (us)
# [MLU IO Efficiency      ]: 0.23275
# [MLU Compute Efficiency ]: 0.5
# [Mlu Workspace Size     ]: -1 (Bytes)
# 
# ----------- case1 -----------
# ...
```

Platform：MLU590
```bash
# ----------- case0 -----------
# ----------- case1 -----------
# ...
```

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.
precheckin 实际已通过

### CODE REVIEW

File: docs/design_docs/poly_nms/poly_nms.md [hunk 1]
```diff
@@ -131,7 +131,6 @@ dets = [[0, 0, 2, 0, 2, 2, 0, np.nan, 2], [1.5, 1.5, 2.5, 1.5, 2.5, 2.5, 1.5, 2.
 |  原位限制     | 不支持原位                                                 |
 | stride限制   | 不支持stride                                      |
 | 广播限制     | 不支持广播                                                   |
-| 规模限制     | mlu270,mlu290及mlu370上输入boxes个数不超过9770个，超过规模限制会有打印报错日志。|
```
Comment (chqy99) [inline docs/design_docs/poly_nms/poly_nms.md:134]: 370 有限制的话，只需要删除 “mlu270,mlu290及”

**Review by chqy99 — APPROVED** [review]

File: mlu_op.h [hunk 2]
```diff
@@ -6721,7 +6721,6 @@ mluOpThreeInterpolateForward(mluOpHandle_t handle,
  * @par Note
  * - The value of \b indices must be in the range of [0, M-1], otherwise the output result
  *   is meaningless and the corresponding output will be set to 0.
- * - In MLU270 and MLU290, the maximum value in the \b indices should be less than
  *   2^23, otherwise the output result is not guaranteed to be correct.
```
Comment (PetrelYy) [inline mlu_op.h:6724]: 少删一行？

File: docs/design_docs/poly_nms/poly_nms.md (hunk 1 above)
Comment (DanieeelLiu) [reply docs/design_docs/poly_nms/poly_nms.md:134]: Fixed

File: mlu_op.h (hunk 2 above)
Comment (DanieeelLiu) [reply mlu_op.h]: Fixed

**Review by PetrelYy — APPROVED** [review]

**Review by ArtIntAI — APPROVED** [review]

File: mlu_op.h [hunk 3]
```diff
@@ -4637,7 +4631,6 @@ mluOpGetNmsWorkspaceSize(mluOpHandle_t handle,
  * - The shape[0] of the \b output should be equal to the input height.
  * - The shape[1] of the \b output should be equal to the input width.
  * - The shape[2] of the \b output and \b var must be less than 2100
- *   in MLU200 series, and less than 2900 in MLU300 series.
```
Comment (AndyQiao0828) [inline mlu_op.h:4640]: 2100》》2900

File: mlu_op.h [hunk 4]
```diff
@@ -5112,8 +5105,8 @@ mluOpDestroyRoiAlignForwardDescriptor(mluOpRoiAlignForwardDescriptor_t desc);
  * - This function should be called with ::mluOpSetRoiAlignForwardDescriptor_v2.
  *
  * @par Note
- * - When \b input contains NaN. If \b pool_mode is maximum pooling_mode, \b output is positive
- *   saturation value on MLU200 series, and \b output gets more NaN than ieee754 on MLU300 series.
+ * - When \b input contains NaN. If \b pool_mode is maximum pooling_mode, \b output gets more NaN than
+ *   ieee754 on MLU300 series.
```
Comment (AndyQiao0828) [inline mlu_op.h:5109]:
```suggestion
 * - When \b input contains NaN, if \b pool_mode is maximum pooling_mode, \b output gets more NaN than
 *   IEEE 754 on MLU300 series.
```

File: mlu_op.h [hunk 5]
```diff
@@ -6104,8 +6097,7 @@ mluOpVoxelization(mluOpHandle_t handle,
  * - The third dimension of scores tensor must be equal to \b class_num.
  * - The fourth dimension of boxes tensor and scores tensor must be equal to the
  *   multiplication result of the third dimension and the fourth dimension of input x tensor.
- * - The \b class_num should be larger than 0. On MLU200, the value cannot be
- *   greater than 1534. On MLU300, the value cannot be greater than 2558.
+ * - The \b class_num should be larger than 0. On MLU300, the value cannot be greater than 2558.
```
Comment (AndyQiao0828) [inline mlu_op.h:6100]:
```suggestion
 * - The \b class_num should be larger than 0. On MLU300 series, the value cannot be greater than 2558.
```

File: mlu_op.h [hunk 6]
```diff
@@ -11499,19 +11471,14 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
  * - \b Output consists of [rois_num, pooled_h, pooled_w, channels]. In the dimensions of the h and w of the input
  *   and the output, (\b x2 - \b x1) * (\b y2 - \b y1) * \b spatial_scale * \b spatial_scale / (\b pooled_h * \b
  *   pooled_w) < (nram_limitation / 32). Nram_limitation means the limitation of the nram. When the supported MLU
- *   platform is 200, the nram_limitation is (98304 - 4 * \b channels) / 2. When the supported MLU platform is 300,
- *   the nram_limitation is (163804 - 4 * \b channels) / 2. \b pooled_h means height of output. \b pooled_w means width
- *   of output.
+ *   platform is 300, the nram_limitation is (163804 - 4 * \b channels) / 2. \b pooled_h means height of output.
+ *   \b pooled_w means width of output.
```
Comment (AndyQiao0828) [inline mlu_op.h:11475]: When the supported MLU platform is 300,》》On MLU300 series,  改成这样？

**Review by stulai — APPROVED** [review]

File: mlu_op.h (hunk 3 above)
Comment (DanieeelLiu) [reply mlu_op.h]: Fixed

File: mlu_op.h (hunk 4 above)
Comment (DanieeelLiu) [reply mlu_op.h]: Fixed

File: mlu_op.h (hunk 5 above)
Comment (DanieeelLiu) [reply mlu_op.h]: Fxied

File: mlu_op.h (hunk 6 above)
Comment (DanieeelLiu) [reply mlu_op.h]: Fixed

**Review by AndyQiao0828 — APPROVED** [review]

**Review by aokbok — APPROVED** [review]

**Pull request merged**